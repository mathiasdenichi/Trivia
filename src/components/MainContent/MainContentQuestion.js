import React from 'react'
import PropTypes from 'prop-types'

const Question = ({ question }) => (
  <div
    key={`${question}-questions`}
    id='questionBox'
    name='questionBox'
  >
    <p>
      {question}
    </p>
  </div>
)

Question.propTypes = {
  question: PropTypes.string,
}
Question.defaultProps = {
  question: '',
}

export default Question
