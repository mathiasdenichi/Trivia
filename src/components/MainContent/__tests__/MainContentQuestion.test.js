import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainContentQuestion from '../MainContentQuestion'

configure({ adapter: new Adapter() })


describe('Main Content Question', () => {
  const wrapper = shallow(<MainContentQuestion />)
  it('renders the component and matches it to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
