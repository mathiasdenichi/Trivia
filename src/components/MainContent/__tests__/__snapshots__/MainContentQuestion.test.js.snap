// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Content Question renders the component and matches it to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    question=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "-questions",
    "nodeType": "host",
    "props": Object {
      "children": <p>
        
      </p>,
      "id": "questionBox",
      "name": "questionBox",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "-questions",
      "nodeType": "host",
      "props": Object {
        "children": <p>
          
        </p>,
        "id": "questionBox",
        "name": "questionBox",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
